ev3dev2/motor.py

```diff
@@ -2551,51 +2595,144 @@ def _on_arc(self, speed, radius_mm, distance_mm, brake, block, arc_right):
+                degrees_error = angle_current_degrees - angle_target_degrees
+
+            # We need to rotate counter-clockwise
+            else:
+                degrees_error = (angle_target_degrees - angle_current_degrees) * -1
+
+            log.info("%s: _turn() ended up at %s, error %s, wiggle_room %s" %
+               (self, angle_current_degrees, degrees_error, wiggle_room))
+
+            if abs(degrees_error) > wiggle_room:
+                self._turn(speed, degrees_error, brake, block)
+
+    def turn_right(self, speed, degrees, brake=True, block=True, wiggle_room=2):
         """
-        Rotate clockwise 'degrees' in place
+        Rotate clockwise `degrees` in place
         """
-        self._turn(speed, abs(degrees), brake, block)
+        self._turn(speed, abs(degrees), brake, block, wiggle_room)
 
-    def turn_left(self, speed, degrees, brake=True, block=True):
+    def turn_left(self, speed, degrees, brake=True, block=True, wiggle_room=2):
         """
-        Rotate counter-clockwise 'degrees' in place
+        Rotate counter-clockwise `degrees` in place
         """
-        self._turn(speed, abs(degrees) * -1, brake, block)
+        self._turn(speed, abs(degrees) * -1, brake, block, wiggle_room)
+
+    def turn_degrees(self, speed, degrees, brake=True, block=True, wiggle_room=2):
+        """
+        Rotate `degrees` in place
+        """
+        if degrees > 0:
+            self.turn_right(speed, degrees, brake, block, wiggle_room)
+        elif degrees < 0:
+            self.turn_left(speed, abs(degrees), brake, block, wiggle_room)
+
+    def turn_to_angle(self, speed, angle_target_degrees, brake=True, block=True, wiggle_room=2):
+        """
+        Rotate in place to `angle_target_degrees` at `speed`
+        """
+        assert self.odometry_thread_run, "odometry_start() must be called to track robot coordinates"
+
+        # Make both target and current angles positive numbers between 0 and 360
+        while angle_target_degrees < 0:
+            angle_target_degrees += 360
+
+        angle_current_degrees = math.degrees(self.theta)
+
+        while angle_current_degrees < 0:
+            angle_current_degrees += 360
+
+        # Is it shorter to rotate to the right or left
+        # to reach angle_target_degrees?
+        if angle_current_degrees > angle_target_degrees:
+            turn_right = True
+            angle_delta = angle_current_degrees - angle_target_degrees
+        else:
+            turn_right = False
+            angle_delta = angle_target_degrees - angle_current_degrees
+
+        if angle_delta > 180:
+            angle_delta = 360 - angle_delta
+            turn_right = not turn_right
+
+        log.debug("%s: turn_to_angle %s, current angle %s, delta %s, turn_right %s" %
+            (self, angle_target_degrees, angle_current_degrees, angle_delta, turn_right))
+        self.odometry_coordinates_log()
+
+        if turn_right:
```

Comment:
Similar to above, I think we should avoid the special-casing, both for readability and potential minor performance gains.